framework/test/unit/modules/random/endpoint.spec.ts

```diff
@@ -255,6 +295,123 @@ describe('RandomModuleEndpoint', () => {
 			);
 		});
 
+		it('should throw error when distance is greater than count', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const seed = '7c73f00f64fcebbab49a6145ef14a843';
+			const count = 1000;
+			const distance = 1001;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Invalid distance. Distance must be less than or equal to count',
+			);
+		});
+
+		it('should throw error when hashes is provided but not count', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const distance = 1001;
+			context.params = { address, hashes: ['7c73f00f64fcebbab49a6145ef14a843'], distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Hashes must be provided with count and distance.',
+			);
+		});
+
+		it('should throw error when hashes is provided but not distance', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const count = 1000000;
+			context.params = { address, hashes: ['7c73f00f64fcebbab49a6145ef14a843'], count };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Hashes must be provided with count and distance.',
+			);
+		});
+
+		it('should throw error when hashes empty', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const count = MAX_HASH_COMPUTATION * 10;
+			context.params = {
+				address,
+				hashes: ['7c73f00f64fcebbab49a6145ef14a843', '7c73f00f64fcebbab49a6145ef14a84300'],
+				count,
+				distance: 1,
+			};
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'must NOT have more than 32 characters',
+			);
+		});
+
+		it('should throw error when count is not multiple of distance', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const count = MAX_HASH_COMPUTATION * 10;
```

Comment:
To make the test focused, how about decreasing the count to `MAX_HASH_COMPUTATION`?

And then also adding a separate test which passes when `count > MAX_HASH_COMPUTATION` and hashes are provided?